src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -714,6 +724,66 @@ private string ReadNonceCookie(string nonce)
             return null;
         }
 
+        private void GenerateCorrelationId([NotNull] AuthenticationProperties properties)
+        {
+            var correlationKey = OpenIdConnectAuthenticationDefaults.CookieStatePrefix;
+
+            var nonceBytes = new byte[32];
+            CryptoRandom.GetBytes(nonceBytes);
+            var correlationId = TextEncodings.Base64Url.Encode(nonceBytes);
+
+            var cookieOptions = new CookieOptions
+            {
+                HttpOnly = true,
+                Secure = Request.IsHttps,
+                Expires = DateTime.UtcNow + Options.ProtocolValidator.NonceLifetime
```

Comment:
State and nonce are two different things, but I guess using the same lifetime is not a bad thing.